### PR TITLE
zephyr: add minimal footprint configuration for nrf52

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added Kconfig project configuration whit shows the
minimal flash footprint on nRF2832 SoC targets.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: boot/zephyr/boards/nrf52_minimal_footprint.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Minima MCUBoot flash footprint configuration
```
Comment (utzig) [inline boot/zephyr/boards/nrf52_minimal_footprint.conf]: typo: Minimal

File: boot/zephyr/boards/nrf52_minimal_footprint.conf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# for nRF52832 SoC targets
+
+# Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib)
+CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256=y
+CONFIG_BOOT_SIGNATURE_KEY_FILE="root-ec-p256.pem"
+# CONFIG_BOOT_VALIDATE_SLOT0 is not set
```
Comment (utzig) [inline boot/zephyr/boards/nrf52_minimal_footprint.conf]: Although this might not be minimal I would strongly encourage people to have it set! :-P

**Comment by nvlsianpu** [comment]
This is configuration I used as base for different revision comparison recently.

File: boot/zephyr/boards/nrf52_minimal_footprint.conf (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: It is the minimal, not the recommend one. Will consider addition of a comment on implication.

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: added comment.

File: boot/zephyr/boards/nrf52_minimal_footprint.conf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Minimal MCUBoot flash footprint configuration
+# for nRF52832 SoC targets
+# This is not recomendet configuration because of security and reliability
+# reasons.
+
```
Comment (carlescufi) [inline boot/zephyr/boards/nrf52_minimal_footprint.conf]: You are missing quite a few from:
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/basic/minimal/common.conf
and
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/basic/minimal/no-timers.conf

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: #CONFIG_SYS_CLOCK_EXISTS=y is required for reboot routine.

I added the rest form minimal - it has no impact on footprint as configuration was by default the same or an API was no used.

Comment (carlescufi) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: > #CONFIG_SYS_CLOCK_EXISTS=y is required for reboot routine.

Really? That's a shame, are you sure we cannot reboot without a timer? Seems counter intuitive to me.

> I added the rest form minimal - it has no impact on footprint as configuration was by default the same or an API was no used.

It has no impact on the footprint? That's strange, there were a bunch there that I know reduced the code size.

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: looks like only zephyr API which requires CONFIG_SYS_CLOCK_EXISTS is `sys_clock_disable()` which is called before jump to the application. Can do this optional - but probably the call is for something here,

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: https://github.com/JuulLabs-OSS/mcuboot/blob/master/boot/zephyr/main.c#L77

Comment (carlescufi) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: Can you then change that code to:

```
#ifdef(CONFIG_SYS_CLOCK_EXISTS)
sys_clock_disable();
#endif
```

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf52_minimal_footprint.conf]: Yes - actually I was about asking whether this is good idea.

**Pull request merged**